source/thunkify.js

```diff
@@ -0,0 +1,30 @@
+import _arity from './internal/_arity';
+import _curry1 from './internal/_curry1';
+
+/**
+ * Creates a thunk out of a function. A thunk delays a calculation until
+ * its result is needed, providing lazy evaluation of arguments.
+ *
+ * @func
+ * @memberOf R
+ * @category Function
+ * @sig ((a, b, ..., j) -> k) -> (a, b, ..., j) -> (() -> k)
+ * @param {Function} fn A function to wrap in a thunk
+ * @return {Function} Expects arguments for `fn` and returns a new function
+ *  that, when called, applies those arguments to `fn`.
+ * @see R.partial, R.partialRight
+ * @example
+ *
+ *      R.thunkify(R.identity)(42)(); //=> 42
+ *      R.thunkify((a, b) => a + b)(25, 17)(); //=> 42
+ */
+var thunkify = _curry1(function thunkify(fn) {
+  return _arity(fn.length, function createThunk() {
```

Comment:
Can we switch this to `curryN` rather than `arity`?

Reply:
Will do.